storage/aptosdb/src/db/include/aptosdb_reader.rs

```diff
@@ -525,10 +525,6 @@ impl DbReader for AptosDB {
     fn get_block_timestamp(&self, version: u64) -> Result<u64> {
         gauged_api("get_block_timestamp", || {
             self.error_if_ledger_pruned("NewBlockEvent", version)?;
-            ensure!(
-                version <= self.get_latest_version()?,
-                "version older than latest version"
-            );
```

Comment:
removed since the underlying call checks the version (and it rightfully check the committed version instead of the synced version instaead)